### PR TITLE
Implement tuple for family argument for modify_config_set_family_selection

### PR DESCRIPTION
**Pull request by saberlynx**
**What**:

Allow to select auto add nvts for each family individually.

**Why**:

It wasn't possible to set this for all families in one go.

**How**:

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: gvm/protocols/gmpv7/gmpv7.py [hunk 1]
```diff
@@ -4770,10 +4770,9 @@ def modify_config_set_nvt_selection(
     def modify_config_set_family_selection(
         self,
         config_id: str,
-        families: List[str],
+        families: List[Tuple[str, bool]],
         *,
         auto_add_new_families: Optional[bool] = True,
```
Comment (y0urself) [inline gvm/protocols/gmpv7/gmpv7.py:4775]:
```suggestion
        auto_add_new_families: Optional[bool] = True
```

File: gvm/protocols/gmpv9/gmpv9.py [hunk 2]
```diff
@@ -801,10 +801,9 @@ def modify_policy_set_nvt_selection(
     def modify_policy_set_family_selection(
         self,
         policy_id: str,
-        families: List[str],
+        families: List[Tuple[str, bool]],
         *,
         auto_add_new_families: Optional[bool] = True,
```
Comment (y0urself) [inline gvm/protocols/gmpv9/gmpv9.py:806]:
```suggestion
        auto_add_new_families: Optional[bool] = True
```

Comment (y0urself) [reply gvm/protocols/gmpv9/gmpv9.py:806]: This will fix 3.5 I guess.

Comment (saberlynx) [reply gvm/protocols/gmpv9/gmpv9.py:806]: Thanks!

**Review by y0urself — APPROVED** [review]

File: gvm/protocols/gmpv7/gmpv7.py [hunk 3]
```diff
@@ -4783,8 +4782,6 @@ def modify_config_set_family_selection(
             families: List of NVT family names to select.
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:4782]: The comment needs to be updated too

File: gvm/protocols/gmpv9/gmpv9.py [hunk 4]
```diff
@@ -814,14 +813,11 @@ def modify_policy_set_family_selection(
             families: List of NVT family names to select.
```
Comment (bjoernricks) [inline gvm/protocols/gmpv9/gmpv9.py:813]: Same here

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Looks good! Just the comments for the families parameters should be updated too

File: gvm/protocols/gmpv7/gmpv7.py (hunk 3 above)
Comment (saberlynx) [reply gvm/protocols/gmpv7/gmpv7.py:4782]: True. Do you think we need to set a default True for this too? like auto_add_new_families?

Comment (bjoernricks) [reply gvm/protocols/gmpv7/gmpv7.py:4782]: Currently I don't know how that should be possible easily. In that case you would need to check each item in the list if it is a tuple and if the second value in the tuple is available.

Comment (y0urself) [reply gvm/protocols/gmpv7/gmpv7.py:4782]: wouldn't be that much work, I guess?

Comment (y0urself) [reply gvm/protocols/gmpv7/gmpv7.py:4782]: You should at least test for tuples like `('foo', )` and `('foo', None)` and `('foo', 'bar')` ... and throw some InvalidArgument errors?

**Review by y0urself — CHANGES_REQUESTED** [review]

**Review by y0urself — APPROVED** [review]

**Pull request merged**